File: Semester 2/APO C++/Tests/V 1.2.0/_TUA/T_Tests_A.cpp
//
// IUT de Nice / Departement informatique / Module APO-C++
// Annee 2009_2010 - Charte des tests unitaires
//
// Exemple d'un module de tests unitaires avec cas d'anomalies
//
// Auteur : A. Thuaire
//
#include "..\_SRC\Tests.h"

void main () {
  
   Tests::Begin("Tests", "1.2.0");

      Tests::Design("Visualisation de niveau 0", 0);
         
         Tests::Case("C1 : 2 / OK"); {			
            Tests::Unit("x", "x");
            Tests::Unit(0, 0);
         }
      			       
      Tests::Design("Visualisation de niveau 1", 1);		
  
	 Tests::Case("C1 : 1 / OK"); {
            Tests::Unit("niveau 1", "niveau 1");
            Tests::Unit(1, 1);
         }

      Tests::Design("Visualisation de niveau 2", 2);		
  
         Tests::Case("C1 : 2 / OK"); {
            Tests::Unit("xxx", "xxx");
            Tests::Unit(2, 2);
         }

         Tests::Case("C2 : 3 / OK"); {
            Tests::Unit("niveau 1", "niveau 1");
            Tests::Unit(1, 1);
            Tests::Unit(-1,-1); 
         } 

      Tests::Design("Visualisation de niveau 3", 3);		
  
	 Tests::Case("C1 : 3 / OK"); {
            Tests::Unit("......", "......");
            Tests::Unit(3, 3);
            Tests::Unit(3, 3);
         }

         Tests::Case("C2 : 2 / OK"); {
            Tests::Unit("xxxxx", "xxxxx");
            Tests::Unit(-3, -3); 
         }
            
         Tests::Case("C3 : 1 / OK"); {
            Tests::Unit(-3.0f, -3.0f);
         }  
      
      Tests::Design("Visualisation de niveau 3 avec TU NOK", 3);		
  
         Tests::Case("C1 : 2 / OK"); {
            Tests::Unit("......", "......");
            Tests::Unit(4, 4);
         }

         Tests::Case("C2 : 1 / NOK !"); {
            Tests::Unit(-4.0f, -4.01f); 
         }
            
         Tests::Case("C3 : 1 / OK"); {
            Tests::Unit(-4.25f, -4.25f);
         }
             
	Tests::End();
}

File: Semester 2/APO_Java/Documents_techniques/Tests_unitaires/Tests/V 2.2.0/T_Tests_A.java
/**
*
* Annee 2009_2010 - Module Java - Charte des tests unitaires
*
* Classe Tests - Module de tests unitaires (cas d'anomalies)
*
* @author Alain Thuaire
*
**/

public class T_Tests_A {

   public static void main (String[] args) {
        
      Tests.Begin("Tests", "2.2.0");

      Tests.Design("Visualisation de niveau 0", 0);
         
         Tests.Case("C1 : 1 TU NOK");			
            Tests.Unit("x", "y");
            Tests.Unit(new Integer(0), new Integer(0));

         Tests.Case("C2 : 2 TU / Second TU NOK");			
            Tests.Unit("x", "x");
            Tests.Unit(new Integer(0), new Integer(1)); 
                			       
      Tests.Design("Visualisation de niveau 1", 1);		
  
         Tests.Case("C1 : 1 TU NOK");
            Tests.Unit("niveau 1", "niveau 2");

      Tests.Design("Visualisation de niveau 2", 2);		
  
         Tests.Case("C1 : 2 TU OK");
            Tests.Unit("xxx", "xxx");
            Tests.Unit(new Integer(2), new Integer(2));

         Tests.Case("C2 : 3 TU / Troisieme TU NOK");
            Tests.Unit("niveau 1", "niveau 1");
            Tests.Unit(new Integer(1), new Integer(1));
            Tests.Unit(new Integer(-1), new Integer(-2));  

         Tests.Case("C3 : 2 TU OK");
            Tests.Unit("yyy", "yyy");
             
      Tests.Design("Visualisation de niveau 3", 3);		
  
         Tests.Case("C1 : 3 TU OK");
            Tests.Unit("......", "......");
            Tests.Unit(new Integer(3), new Integer(3));
            Tests.Unit(new Float(3), new Float(3));

         Tests.Case("C2 : 2 TU OK");
            Tests.Unit("xxxxx", "xxxxx");
            Tests.Unit(new Integer(-3), new Integer(-3)); 
            
         Tests.Case("C3 : 1 TU NOK");
            Tests.Unit(new Float(-3), new Float(3));
            
      Tests.Design("Traitement des references nulles", 3);		
  
         Tests.Case("Un seul parametre est null");
            Tests.Unit("...", null);
            Tests.Unit(null, "...");
                     
      Tests.End();

   }
}
